src/extra_metas.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    crate::state::{SEED_PREFIX_HOLDER_REWARDS, SEED_PREFIX_HOLDER_REWARDS_POOL},
+    solana_program::program_error::ProgramError,
+    spl_tlv_account_resolution::{account::ExtraAccountMeta, seeds::Seed},
+};
+
+/// Extra account metas required by the Paladin Rewards program's SPL Transfer
+/// Hook Interface implementation.
+///
+/// Accounts required (* = extra meta):
+///
+/// 0. `[ ]` Source token account.
+/// 1. `[ ]` Token mint.
+/// 2. `[ ]` Destination token account.
+/// 3. `[ ]` Source owner.
+/// 4. `[ ]` Extra account metas account.
+/// 5. `[ ]` * Holder rewards pool account.
+/// 6. `[w]` * Source holder rewards account.
+/// 7. `[w]` * Destination holder rewards account.
+pub fn get_extra_account_metas() -> Result<[ExtraAccountMeta; 3], ProgramError> {
```

Comment:
nit: If this returns an error, it looks like it would be a programmer error since it takes no input. It would probably be better to `unwrap` / `expect` everything and return the metas directly

Reply:
Bahh I originally had it setup as a static `EXTRA_METAS` using `lazy_static` but the `unwrap` was making me nervous. You think this is acceptable then?

Reply:
Actually I noticed added `lazy_static` and using it for a lazy item in the program processor causes the ELF parsing to fail. Every invocation returns `Program is not deployed`.

Reply:
I just went with `unwrap()` and didn't use `lazy_static`. Weird that it does that, though.

Reply:
lazy_static still needs to be computed at some point when the code is *running*, and not during compile-time, which makes it a lot like mutable static data, which isn't allowed in solana programs. if it were `const` then this could work, but it requires every function to also be const, but const rust is not a super fun programming experience